ipv-core/src/main/java/com/ibm/whc/deid/util/ICDv9Manager.java

```diff
@@ -1,116 +1,136 @@
 /*
- * (C) Copyright IBM Corp. 2016,2020
+ * (C) Copyright IBM Corp. 2016,2021
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 package com.ibm.whc.deid.util;
 
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.Serializable;
+import java.security.SecureRandom;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.HashMap;
-
 import org.apache.commons.csv.CSVParser;
 import org.apache.commons.csv.CSVRecord;
-
 import com.ibm.whc.deid.models.ICD;
 import com.ibm.whc.deid.models.ICDFormat;
+import com.ibm.whc.deid.models.ICDWithoutFormat;
 import com.ibm.whc.deid.shared.localization.Resource;
 import com.ibm.whc.deid.shared.localization.Resources;
 import com.ibm.whc.deid.util.localization.LocalizationManager;
 import com.ibm.whc.deid.util.localization.ResourceEntry;
 import com.ibm.whc.deid.utils.log.LogCodes;
 import com.ibm.whc.deid.utils.log.LogManager;
 
-/** The type Ic dv 9 manager. */
-public class ICDv9Manager implements Manager, Serializable {
-  /** */
-  private static final long serialVersionUID = -5865782878844917863L;
-
-	protected final Collection<ResourceEntry> resourceICDList;
+/** The type ICDv9Manager. */
+public class ICDv9Manager implements Manager {
 
-  protected final MapWithRandomPick<String, ICD> icdByCodeMap;
-  protected final MapWithRandomPick<String, ICD> icdByNameMap;
+  private static final LogManager logger = LogManager.getInstance();
 
-  private static LogManager logger = LogManager.getInstance();
-  protected final Resources resourceType = Resource.ICDV9;
+  protected static final Resources resourceType = Resource.ICDV9;
 
-  protected final String tenantId;
+  private final SecureRandom random = new SecureRandom();
 
-  	/**
-	 * Instantiates a new Ic dv 9 manager.
-	 *
-	 * @param tenantId
-	 * @paramlocalizationProperty location of the localization property file
-	 */
-	public ICDv9Manager(String tenantId, String localizationProperty) {
-    this.tenantId = tenantId;
-		resourceICDList = LocalizationManager.getInstance(localizationProperty)
-				.getResources(Resource.ICDV9);
-    this.icdByCodeMap = new MapWithRandomPick<>(new HashMap<String, ICD>());
-    this.icdByNameMap = new MapWithRandomPick<>(new HashMap<String, ICD>());
+  private final ArrayList<ICDWithoutFormat> icdList = new ArrayList<>(14500);
+  private final HashMap<String, ICDWithoutFormat> icdByCodeMap = new HashMap<>(22000);
+  private final HashMap<String, ICDWithoutFormat> icdByNameMap = new HashMap<>(22000);
+  private final HashMap<String, ICDWithoutFormat> icdByShortMap = new HashMap<>(22000);

```

Comment:
Same comment as above regarding the "hardcoded starting values" .

Reply:
Yes - see comment above